src/main/java/org/prebid/server/handler/ForceDealsUpdateHandler.java

```diff
@@ -0,0 +1,119 @@
+        this.lineItemService = Objects.requireNonNull(lineItemService);
+        this.endpoint = Objects.requireNonNull(endpoint);
+    }
+
+    @Override
+    public void handle(RoutingContext routingContext) {
+        try {
+            handleDealsAction(dealsActionFrom(routingContext));
+            HttpUtil.executeSafely(routingContext, endpoint,
+                    response -> response
+                            .setStatusCode(HttpResponseStatus.NO_CONTENT.code())
+                            .end());
+        } catch (InvalidRequestException e) {
+            respondWithError(routingContext, HttpResponseStatus.BAD_REQUEST, e);
+        } catch (Exception e) {
+            respondWithError(routingContext, HttpResponseStatus.INTERNAL_SERVER_ERROR, e);
+        }
+    }
+
+    private void handleDealsAction(DealsAction dealsAction) {
+        switch (dealsAction) {
+            case UPDATE_LINE_ITEMS:
+                plannerService.updateLineItemMetaData();
+                break;
+            case SEND_REPORT:
+                deliveryStatsService.sendDeliveryProgressReports();
+                break;
+            case REGISTER_INSTANCE:
+                registerService.performRegistration();
+                break;
+            case RESET_ALERT_COUNT:
+                alertHttpService.resetAlertCount("pbs-register-client-error");
+                alertHttpService.resetAlertCount("pbs-planner-client-error");
+                alertHttpService.resetAlertCount("pbs-planner-empty-response-error");
+                alertHttpService.resetAlertCount("pbs-delivery-stats-client-error");
+                break;
+            case CREATE_REPORT:
+                deliveryProgressService.createDeliveryProgressReports(ZonedDateTime.now());
+                break;
+            case INVALIDATE_LINE_ITEMS:
+                lineItemService.invalidateLineItems();
+                break;
+            default:
+                throw new IllegalStateException("Unexpected action value");
+        }
+    }
+
+    private DealsAction dealsActionFrom(RoutingContext routingContext) {
+        final String givenActionValue = routingContext.request().getParam(ACTION_NAME_PARAM);
+        if (StringUtils.isEmpty(givenActionValue)) {
+            throw new InvalidRequestException(String.format(
+                    "Parameter '%s' is required and can't be empty",
+                    ACTION_NAME_PARAM));
+        }
+
+        try {
+            return DealsAction.valueOf(givenActionValue.toUpperCase());
+        } catch (IllegalArgumentException ignored) {
+            throw new InvalidRequestException(String.format(
+                    "Given '%s' parameter value '%s' is not among possible actions",
+                    ACTION_NAME_PARAM,
+                    givenActionValue));
+        }
+    }
+
+    private void respondWithError(RoutingContext routingContext, HttpResponseStatus statusCode, Exception exception) {
```

Comment:
Can be static.

Reply:
It can't as uses 'endpoint' field from non-static context